docs/eventing/sources/pingsource.md

```diff
@@ -1,196 +1,261 @@
+EOF
+```
+{{< /tab >}}
 
----
+{{% tab name="kn" %}}
+Create the service using the kn command-line interface by running:
 
-kind: Service
-apiVersion: v1
-metadata:
-  name: event-display
-spec:
-  selector:
-    app: event-display
-  ports:
-  - protocol: TCP
-    port: 80
-    targetPort: 8080
-EOF
+```shell
+kn service create event-display --image gcr.io/knative-releases/knative.dev/eventing-contrib/cmd/event_display
 ```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Create a producer (PingSource)
 
-### Creating the PingSource
+Create a producer (PingSource) that sends events to your consumer every two minutes.
+The `sink` element in the metadata describes where to send events.
+In this case, events are sent to a service with the name `event-display`.
+This is a tight connection between the producer and consumer.
 
-You can now create the `PingSource` sending an event containing
-`{"message": "Hello world!"}` every minute.
+For each set of ping events that you want to request, create an event source in the same namespace
+as the destination.
 
 {{< tabs name="create-source" default="YAML" >}}
 {{% tab name="YAML" %}}
+Create the event source from stdin by running:
 
 ```shell
-kubectl create -n pingsource-example -f - <<EOF
-apiVersion: sources.knative.dev/v1beta2
+cat <<EOF | kubectl create -f -
+apiVersion: sources.knative.dev/v1alpha2
 kind: PingSource
 metadata:
   name: test-ping-source
 spec:
-  schedule: "*/1 * * * *"
+  schedule: "*/2 * * * *"
   contentType: "application/json"
   data: '{"message": "Hello world!"}'
   sink:
     ref:
-      apiVersion: v1
+      apiVersion: serving.knative.dev/v1
       kind: Service
       name: event-display
 EOF
 ```
-
 {{< /tab >}}
 
 {{% tab name="kn" %}}
-Notice that the namespace is specified in two places in the command in `--namespace` and the `--sink` hostname
+Create the event source from the `ping-source.yaml` file by running:
+
 ```shell
 kn source ping create test-ping-source \
-  --namespace pingsource-example \
-  --schedule "*/1 * * * *" \
-  --data '{"message": "Hello world!"}' \
-  --sink http://event-display.pingsource-example.svc.cluster.local
+  --schedule "*/2 * * * *" --data \
+  '{ "message": "Hello world!" }' \
+  --sink ksvc:event-display
 ```
-
 {{< /tab >}}
 {{< /tabs >}}
 
 ## (Optional) Create a PingSource with binary data
 
-Sometimes you may want to send binary data, which cannot be directly serialized in yaml, to downstream. This can be achieved by using `dataBase64` as the payload. As the name suggests, `dataBase64` should carry data that is base64 encoded.
+You might sometimes want to send binary data, which cannot be directly serialized in YAML, to
+downstream. You can do so using `dataBase64` as the payload.
+As the name suggests, `dataBase64` should carry data that is base64-encoded.
 
-Please note that `data` and `dataBase64` cannot co-exist.
+**Note:** `data` and `dataBase64` cannot co-exist.
```

Comment:
```suggestion
**NOTE:** You can use either `data` or `dataBase64`, but not both.
```
Explain why?

Reply:
@lionelvillard, why can't you use both?

Reply:
ping @lionelvillard

Reply:
Both `data` and `dataBase64` specify the CloudEvent payload, that's why those fields are mutually-exclusive. 

Does that answer your question?

Reply:
Maybe the following text, to clarify why:

```markdown
**NOTE:** Because both `data` and `dataBase64` specify the event payload, only one field may be set.
```